`AppConfig.json` should be stored in XDG configuration directory on Linux

# Improvement

## Currently 
Currently the app seems to attempt to write a `lib/nexus-mods-app/AppConfig.json` within the application directory at startup, specifically when accepting/denying the diagnostics and usage data prompt when launching for the first time.

## We want
Per-user configuration should be stored according to the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html):

>  $XDG_CONFIG_HOME defines the base directory relative to which user-specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used. 

In other words, if `$XDG_CONFIG_HOME` is non-empty, the path should be something like `$XDG_CONFIG_HOME/NexusMods.App/AppConfig.json`. Otherwise the path should be something like `$HOME/.config/NexusMods.App/AppConfig.json`.

## Supporting Information

This causes a runtime crash on NixOS, where applications are installed into a read-only filesystem. After that at least these buttons are unresponsive:

![image](https://github.com/Nexus-Mods/NexusMods.App/assets/168301/17048ecd-a6c4-40c9-be0e-6165aa078b8d)

Log:

```console
❯ ./result/bin/NexusMods.App 
00:00:00.000 [DEBUG] Application starting in release mode
00:00:00.008 [DEBUG] Calling startup handler
00:00:00.009 [INFO] Starting application with args: 
00:00:00.010 [INFO] Starting main process from client process: 534637
00:00:00.012 [DEBUG] Starting main process
00:00:00.012 [INFO] Starting main process /nix/store/3bx00filk7463ng8bd9c9ihkpjpixkpf-nexus-mods-app-0.3/lib/nexus-mods-app/NexusMods.App with arguments: main-process
00:00:00.013 [DEBUG] Main process started with PID: 534651
00:00:00.000 [DEBUG] Application starting in release mode
00:00:00.027 [DEBUG] Calling startup handler
00:00:00.027 [INFO] Starting application with args: main-process
00:00:00.029 [INFO] Main using SyncFile: /run/user/1000/NexusMods.App-single_process.sync
00:00:00.030 [INFO] No main process found, starting main process (534651)
00:00:00.032 [INFO] Started TCP listener on port 33133
00:00:00.032 [INFO] This process (534651) is now the main process
00:00:00.240 [INFO] Found main process 534651 listening on port 33133
00:00:00.057 [INFO] Accepted TCP connection from 45146
00:00:00.244 [DEBUG] Connected to main process 534651 on port 33133
00:00:00.104 [DEBUG] Starting UI window
00:00:00.106 [DEBUG] Using sqlite data store at /home/victor/.local/share/NexusMods.App/DataModel/DataModel.sqlite
00:00:00.619 [INFO] ViewModel is null. Falling back to default content.
00:00:00.620 [INFO] ViewModel is null. Falling back to default content.
00:00:00.621 [INFO] ViewModel is null. Falling back to default content.
00:00:00.621 [INFO] ViewModel is null. Falling back to default content.
00:00:00.621 [INFO] ViewModel is null. Falling back to default content.
00:00:00.636 [WARN] The class NexusMods.App.UI.Overlays.SetOverlayItem property VM is a POCO type and won't send change notifications, WhenAny will only return a single value!
00:00:00.677 [DEBUG] Finding View for NexusMods.App.UI.Controls.TopBar.TopBarViewModel
00:00:00.679 [TRACE] Getting AuthData-4E657875734D6F64732E4E6574776F726B696E672E4E657875735765624170692E4A5754546F6B656E73 of type JWTTokenEntity
00:00:00.681 [TRACE] Getting AuthData-4E657875734D6F64732E4E6574776F726B696E672E4E657875735765624170692E4A5754546F6B656E73 of type JWTTokenEntity
00:00:00.681 [TRACE] Getting AuthData-4E657875734D6F64732E4E6574776F726B696E672E4E657875735765624170692E4A5754546F6B656E73 of type JWTTokenEntity
00:00:00.735 [INFO] Ready to show NexusMods.App.UI.Controls.TopBar.TopBarView with autowired NexusMods.App.UI.Controls.TopBar.TopBarViewModel.
00:00:00.755 [DEBUG] Finding View for NexusMods.App.UI.Controls.Spine.SpineViewModel
00:00:00.756 [INFO] Ready to show NexusMods.App.UI.Controls.Spine.Spine with autowired NexusMods.App.UI.Controls.Spine.SpineViewModel.
00:00:00.757 [DEBUG] Finding View for NexusMods.App.UI.Controls.DevelopmentBuildBanner.DevelopmentBuildBannerViewModel
00:00:00.766 [INFO] Ready to show NexusMods.App.UI.Controls.DevelopmentBuildBanner.DevelopmentBuildBannerView with autowired NexusMods.App.UI.Controls.DevelopmentBuildBanner.DevelopmentBuildBannerViewModel.
00:00:00.770 [WARN] The class ReactiveUI.ReactiveCommand`2[[System.Reactive.Unit, System.Reactive, Version=6.0.0.0, Culture=neutral, PublicKeyToken=94bc3704cddfc263],[System.Reactive.Unit, System.Reactive, Version=6.0.0.0, Culture=neutral, PublicKeyToken=94bc3704cddfc263]] property IsExecuting is a POCO type and won't send change notifications, WhenAny will only return a single value!
00:00:00.771 [DEBUG] Finding View for NexusMods.App.UI.Overlays.MetricsOptIn.MetricsOptInViewModel
00:00:00.774 [INFO] Ready to show NexusMods.App.UI.Overlays.MetricsOptIn.MetricsOptInView with autowired NexusMods.App.UI.Overlays.MetricsOptIn.MetricsOptInViewModel.
00:00:00.859 [DEBUG] Finding View for NexusMods.App.UI.WorkspaceSystem.AddPanelDropDownViewModel
00:00:00.874 [INFO] Ready to show NexusMods.App.UI.WorkspaceSystem.AddPanelDropDownView with autowired NexusMods.App.UI.WorkspaceSystem.AddPanelDropDownViewModel.
00:00:00.882 [DEBUG] Finding View for NexusMods.App.UI.Controls.Spine.Buttons.Icon.IconButtonViewModel
00:00:00.885 [INFO] Ready to show NexusMods.App.UI.Controls.Spine.Buttons.Icon.IconButton with autowired NexusMods.App.UI.Controls.Spine.Buttons.Icon.IconButtonViewModel and contract 'Home'.
00:00:00.887 [INFO] ViewModel is null. Falling back to default content.
00:00:00.891 [DEBUG] Finding View for NexusMods.App.UI.Controls.Spine.Buttons.Download.SpineDownloadButtonViewModel
00:00:00.893 [INFO] Ready to show NexusMods.App.UI.Controls.Spine.Buttons.Download.SpineDownloadButtonView with autowired NexusMods.App.UI.Controls.Spine.Buttons.Download.SpineDownloadButtonViewModel.
00:00:00.964 [INFO] ViewModel is null. Falling back to default content.
00:03:41.003 [INFO] ViewModel is null. Falling back to default content.
00:03:41.024 [ERROR] Unobserved task exception|System.UnauthorizedAccessException: Access to the path '/nix/store/3bx00filk7463ng8bd9c9ihkpjpixkpf-nexus-mods-app-0.3/lib/nexus-mods-app/AppConfig.json' is denied.
 ---> System.IO.IOException: Permission denied
   --- End of inner exception stack trace ---
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirError)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.File.Open(String path, FileMode mode, FileAccess access, FileShare share)
   at NexusMods.Paths.FileSystem.InternalOpenFile(AbsolutePath path, FileMode mode, FileAccess access, FileShare share)
   at NexusMods.Paths.BaseFileSystem.OpenFile(AbsolutePath path, FileMode mode, FileAccess access, FileShare share)
   at NexusMods.Paths.BaseFileSystem.CreateFile(AbsolutePath path)
   at NexusMods.Paths.BaseFileSystem.WriteAllBytesAsync(AbsolutePath path, Byte[] data, CancellationToken cancellationToken)
```

--- a/src/NexusMods.App/AppConfig.cs
+++ b/src/NexusMods.App/AppConfig.cs
@@ -151,10 +151,20 @@ internal class AppConfigManager : IAppConfigManager
     private readonly AbsolutePath _configPath;
     private readonly JsonSerializerOptions _jsonSerializerOptions;
 
+    // TODO: this entire class should be removed once we have settings
+
     public AppConfigManager(AppConfig config, JsonSerializerOptions jsonSerializerOptions)
     {
         _config = config;
-        _configPath = FileSystem.Shared.GetKnownPath(KnownPath.EntryDirectory).Combine("AppConfig.json");
+
+        // On AppImage (Linux), 'OWD' should take precedence over the entry directory if it exists.
+        // https://docs.appimage.org/packaging-guide/environment-variables.html
+        var owd = Environment.GetEnvironmentVariable("OWD");
+
+        _configPath = string.IsNullOrEmpty(owd)
+            ? FileSystem.Shared.GetKnownPath(KnownPath.EntryDirectory).Combine("AppConfig.json")
+            : FileSystem.Shared.FromUnsanitizedFullPath(owd).Combine("AppConfig.json");
+
         _jsonSerializerOptions = jsonSerializerOptions;
     }
 

